announcements.py

```diff
@@ -5,21 +5,35 @@
 import itertools
 from multiprocessing import Process
 
+# Announcement data class for storing the information of each announcement
 class Announcement:
     id_iter = itertools.count()
-    banana_time = datetime.time(10, 30, 0)
+    banana_time = datetime.time(15, 30, 0)
+    # Keeps track of what days the announcements should be sent
+    selected_days = {
+        "monday": True,
+        "tuesday": True,
+        "wednesday": True,
+        "thursday": True,
+        "friday": True,
+        "saturday": False,
+        "sunday": False
+    }
+
     def __init__(self, time, text):
         self.id = next(self.id_iter)
         self.time = time
         self.text = text
 
+# AnnouncementWorker class that makes use of multiprocessing to allow it to run standalone and sleep until the right time
 class AnnouncementWorker(Process):
-    def __init__(self, id, time, text, banana_time):
+    def __init__(self, announcement: Announcement):
         super(AnnouncementWorker, self).__init__()
-        self.id = id
-        self.time = time
-        self.text = text
-        self.banana_time = banana_time
+        self.id = announcement.id
+        self.time = announcement.time
+        self.text = announcement.text
+        self.banana_time = Announcement.banana_time
```

Comment:
Is there any particular reason why banana time and selected days are class variables for Announcement? Is there ever going to be a use case where banana time is going to be different depending on the day of the week?

Reply:
Good question, thanks for bringing it up.  There will only ever be one `banana_time`, so the answer to your second question is no. I opted to put `selected_days` as a class variable for `Announcement` as it is required by the `AnnouncementWorker` classes the same as `banana_time`.

Therefore, by having it as a class variable we need only pass in an individual `Announcement` object instance to the constructor and the worker can initialise itself from there. I think this is a cleaner solution than storing `selected_days` as a global variable in bananabot.py. If you have any alternative suggestions however, please feel free to let me know :)